### PR TITLE
Hint at thread replies when messages are unthreaded

### PR DESCRIPTION
**Pull request by 42wim**
Simpler version of #557

### CODE REVIEW

File: bridge/slack/slack.go [hunk 1]
```diff
@@ -285,6 +285,12 @@ func (b *Bslack) sendRTM(msg config.Message) (string, error) {
 		return "", err
 	}
 
+	// Handle prefix hint for unthreaded messages.
+	if msg.ParentID == "unthreaded" {
+		msg.ParentID = ""
+		msg.Text = fmt.Sprintf("thread reply: %s", msg.Text)
+	}
```
Comment (nylen) [inline bridge/slack/slack.go]: I suggest using the convention/format of `[thread]` here.  It's a bit shorter, and `[prefix]` is less likely to be typed by a user as part of a regular message than `prefix: `.  It's also more consistent with existing machine-generated text like `RemoteNickFormat="[{PROTOCOL}] <{NICK}> "`.

File: gateway/gateway.go [hunk 2]
```diff
@@ -398,6 +398,12 @@ func (gw *Gateway) SendMessage(origmsg config.Message, dest *bridge.Bridge, chan
 		msg.ParentID = canonicalParentMsgID
 	}
 
+	// if the parentID is still empty and we have a parentID set in the original message
+	// this means that we didn't find it in the cache so set it unthreaded
```
[the remark this answers is not in the document]
Comment (patcon) [reply gateway/gateway.go]: I'm down for whatever, but "___thread" words have seemed helpful in speaking about issues: rethreaded, unthreaded, misthreaded, etc -- all things that happen :)

[the remark this answers is not in the document]
Comment (nylen) [reply gateway/gateway.go]: Suggestion after thinking about it a bit more: "message-parent-not-cached"

Precision is good.

[the remark this answers is not in the document]
Comment (42wim) [reply gateway/gateway.go]: what about msg-parent-not-found ? :)

**Comment by ForsakenHarmony** [comment]
Does this only support sending to slack?

**Comment by patcon** [comment]
@ForsakenHarmony yep! or at least if it hasn't been added for other threaded providers (or refactored to be more general) in a more recent PR

**Pull request merged**